lib/src/commands/OptionsProcessor.ts

```diff
@@ -31,6 +31,7 @@ export class OptionsProcessor {
       clone(options),
       (key, parentOptions) => {
         this.deprecations.onProcessOptions(key, parentOptions, commandName);
+        this.deprecations.checkForDeprecatedOptions(parentOptions);
```

Comment:
I think that the original intention with `onProcessOptions` was to actually... to check for deprecated options 😅 
Perhaps we can rename `onProcessOptions` to `checkForDeprecatedOptions`?

Reply:
Do you think that maybe we can leave this as it is and open a ticket for refactoring the deprecation system that can be open to discussions? :D Otherwise I need to think about this alone and maybe others can have some good suggestions.

Reply:
Sure, your call! 👍